src/exported.ts

```diff
@@ -16,6 +16,6 @@ export { Flags } from '@oclif/core';
 
 // custom flags
 export { requiredOrgFlag, requiredHubFlag } from './flags/orgFlags';
-export { buildIdFlag } from './flags/salesforceId';
-export { orgApiVersionFlag as apiVersionFlag } from './flags/apiVersion';
-export { buildDurationFlag, DurationFlagConfig } from './flags/duration';
+export { salesforceIdFlag } from './flags/salesforceId';
```

Comment:
What do you think about exporting this under a single namespace, e.g. `Flags`?

That way we can make these flags feel more like the core flags, e.g.

```typescript
import { SfCommand, Flags } from '@salesforce/sf-plugins-core';

export default class MyCommand {
  static flags = {
    wait: Flags.duration(),
    directory: Flags.directory(),
    'random-string': Flags.string(),
  }
}
```

Reply:
oh, you want people to use the `Flags` from this library instead of from oclif/core directly?  I've mostly seen it the other way.

https://github.com/salesforcecli/plugin-env/blob/main/src/commands/env/display.ts#L8
https://github.com/salesforcecli/plugin-deploy-retrieve-metadata/blob/main/src/commands/deploy/metadata.ts#L9

Reply:
would that conflict with `export { Flags } from '@oclif/core';` (line19)

Reply:
I don't think we have a hard stance on it. But if we're going to provide a lot of specialty flags I like the idea of them being easily discoverable alongside all the built in ones

Reply:
I don't like the pattern of re-exporting stuff from libraries--that's been a lot of pain with command > core where it causes TS problems when the versions aren't lined up.

In this example other plugins would hit`oclif/core.Flags` !== `sf-plugins-core/node_modules/@oclif/core.Flags`

alternatively "just import the flags you need" instead of importing entire namespaces to get a Flag

Reply:
I agree about re-exporting being an anti-pattern. I'm predominately concerned about plugin providers not knowing that these flags exist and so they make their own implementation. In my mind having them all under a single namespace makes them more easily discovered since documentation is often ignored.

Another point: do we really want to bog plugin providers down with multiple ways of getting the flags they need? "Well you can get these flags from oclif but you can also get these other flags from sf-plugins-core... depends on what you need". To me it's a better developer experience to have all the flags together.

Instead of re-exporting `Flags` from oclif, we could do something like this:

```typescript
import { Flags as OclifFlags } from '@oclif/core';
import { salesforceIdFlag } from './flags/salesforceId';

export const Flags = {
  string: OclifFlags.string,
  boolean: OclifFlags.boolean,
  url: OclifFlags.url,
  integer: OclifFlags.integer,
  salesforceIdFlag,
  // etc...
};
```

Reply:
I think that's still re-exporting from TS's perspective, but you're saying it wouldn't bite anyone as long as no plugin imported from oclif/core?  hmm...maybe.

Reply:
I don't think that solution would bite anyone - it's an entirely different export.

Reply:
let's give it a shot.  It could turn out nice if it works.